src/Microsoft.Identity.Client/UiRequiredExceptionClassification.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Details about the cause of an <see cref="MsalUiRequiredException"/>, giving a hint about what the user can expect when 
+    /// they go through interactive authentication. See https://aka.ms/msal-net-UiRequiredException for details.
+    /// </summary>
+    public enum UiRequiredExceptionClassification
```

Comment:
>UiRequiredExceptionClassification [](start = 16, length = 33)

I like the enum, however the name here is challenging. Perhaps this has to stay as ExceptionClassification or we need two enum's one for ui and one for none ui. or this one could be called InvalidGrantClassification :)

Reply:
There are 2 sources of UIRequiredException: 

- client side MSAL logic, such as calling AcquireTokenSilent(string upn) with an invalid upn. CLassificaiton will be `UiRequiredExceptionClassification.None`
- server side logic, where AAD returns an "invalid_grant" exception with an additional sub_error, which is translated to the `UiRequiredExceptionClassification.BasicAction`, UiRequiredExceptionClassification.PasswordExpired` etc. 

@jmprieur - can you help break the naming dispute ?

Reply:
ExceptionClassification ? Or ServiceExceptionClassification?

Reply:
@shoatman - can you also have a look at this please?